src/main/java/org/folio/dao/impl/ConfigurationsDaoImpl.java

```diff
@@ -0,0 +1,210 @@
+    
+    int localLength = results.getResultInfo().getTotalRecords();
+    if(localLength == 1) {
+      return Future.succeededFuture(results.getResults().get(0));
+    } else if (localLength == 0) {
+      if (isPut)
+        return Future.succeededFuture(new SamlConfiguration());
+      else {
+        String warnMessage = "There is an empty DB";
+        LOGGER.warn(warnMessage);
+        return Future.failedFuture(warnMessage);
+      }} else {
+        String errorMessage = String.format("TotalRecords are not unique. The number is : %s", Integer.toString(localLength));
+        LOGGER.error(ERROR_MESSAGE_STRING, errorMessage);
+        return Future.failedFuture(new IllegalArgumentException(errorMessage));
+    }
+  }
+
+   
+  @Override
+  public Future<SamlConfiguration> storeSamlConfiguration(Vertx vertx, OkapiHeaders okapiHeaders, SamlConfiguration samlConfiguration) {
+    
+    Objects.requireNonNull(okapiHeaders);
+    Objects.requireNonNull(samlConfiguration);
+    verifyOkapiHeaders(okapiHeaders);
+    return storeEntry(vertx, okapiHeaders, samlConfiguration);
+  }
+	
+  @Override
+  public Future<SamlConfiguration> storeEntry(Vertx vertx, OkapiHeaders okapiHeaders, String code, String value) {
+    
+    Objects.requireNonNull(okapiHeaders);
+    verifyOkapiHeaders(okapiHeaders);
+
+    return getConfiguration(vertx, okapiHeaders, false)
+      .compose(result -> {
+         switch(code) {
+         case SamlConfiguration.ID_CODE: result.setId(value);
+           break;
+         case SamlConfiguration.IDP_METADATA_CODE: result.setIdpMetadata(value);
+           break;
+         case SamlConfiguration.IDP_URL_CODE: result.setIdpUrl(value);
+           break;
+         case SamlConfiguration.KEYSTORE_FILE_CODE: result.setKeystore(value);
+           break;
+         case SamlConfiguration.KEYSTORE_PASSWORD_CODE: result.setKeystorePassword(value);
+           break;  
+         case SamlConfiguration.KEYSTORE_PRIVATEKEY_PASSWORD_CODE: result.setPrivateKeyPassword(value);
+           break;
+         case SamlConfiguration.METADATA_INVALIDATED_CODE: result.setMetadataInvalidated(value);
+           break;
+         case SamlConfiguration.OKAPI_URL: result.setOkapiUrl(value);
+           break;
+         case SamlConfiguration.SAML_BINDING_CODE: result.setSamlBinding(value);
+           break;
+         case SamlConfiguration.SAML_ATTRIBUTE_CODE: result.setSamlAttribute(value);
+           break;
+         case SamlConfiguration.SAML_CALLBACK: result.setCallback(value);
+           break;  
+         case SamlConfiguration.USER_PROPERTY_CODE: result.setUserProperty(value);
+           break;
+         default: {
+           String errorMessage = String.format("Switch: Incorrect code. The code value is : %s", code);
+           LOGGER.error(ERROR_MESSAGE_STRING, errorMessage);
+           throw new IllegalArgumentException(errorMessage);}
+         }
+         return storeEntry(vertx, okapiHeaders, result);
+        });
+	}
```

Comment:
The indentation is corrected.

Reply:
The changes were published on the 18th of December 2023.

Reply:
The topic of using tabs in my IDE was resolved at the latest  in December 2023. The IDE does not write any tab further. Instead of using an automatic replacement the IDE shows up a special character to mark the existence of a tab. In the file src/test/resources/schemas/XMLSchema.dtd there are 6 tabs contained.

Reply:
The single tab in the file ConfigurationsDaoImpl.java is removed.

Reply:
This means that the file 
https://github.com/folio-org/mod-login-saml/blob/master/src/test/resources/schemas/XMLSchema.dtd of the master branch contains the mentioned tabs.